Library/ViewModels/LiveStreamEventDetailsViewModel.swift

```diff
@@ -0,0 +1,267 @@
+    }.take(first: 1)
+
+    self.subscribeButtonText = subscribed.map {
+      $0 ? Strings.Subscribed() : Strings.Subscribe()
+    }
+
+    self.showErrorAlert = Signal.merge(
+      eventEvent.filter { $0.error != nil }.mapConst(Strings.Failed_to_retrieve_live_stream_event_details()),
+      isSubscribedEvent.filter { $0.error != nil }.mapConst(Strings.Failed_to_update_subscription())
+    )
+
+    self.animateActivityIndicator = Signal.merge(
+      configData.filter { _, _, event in event == nil }.mapConst(true),
+      event.mapConst(false),
+      eventEvent.filter { $0.isTerminating }.mapConst(false),
+      self.showErrorAlert.mapConst(false)
+    ).skipRepeats()
+
+    self.animateSubscribeButtonActivityIndicator = Signal.merge(
+      subscribeIntent.filter { AppEnvironment.current.currentUser != nil }.mapConst(true),
+      self.subscribeButtonTappedProperty.signal
+        .filter { AppEnvironment.current.currentUser != nil }
+        .mapConst(true),
+      isSubscribedEvent.filter { $0.isTerminating }.mapConst(false)
+    ).skipRepeats()
+
+    self.subscribeLabelHidden = Signal.merge(
+      Signal.combineLatest(self.animateSubscribeButtonActivityIndicator, subscribed).map { $0 || $1 },
+      subscribed
+    ).skipRepeats()
+
+    self.detailsStackViewHidden = Signal.merge(
+      self.showErrorAlert.mapConst(true),
+      self.animateActivityIndicator
+    ).skipRepeats()
+
+    self.numberOfPeopleWatchingText = Signal.merge(
+      self.viewDidLoadProperty.signal.mapConst("0"),
+      self.numberOfPeopleWatchingProperty.signal.skipNil().map { Format.wholeNumber($0) }
+      )
+
+    self.subscribeButtonAccessibilityHint = subscribed
+      .map { isSubscribed in
+        isSubscribed
+          ? localizedString(key: "Unsubscribes_from_upcoming_lives_streams.",
+                            defaultValue: "Unsubscribes from upcoming live streams.")
+          : localizedString(key: "Subscribes_to_upcoming_lives_streams",
+                            defaultValue: "Subscribes to upcoming live streams.")
+    }
+
+    self.subscribeButtonAccessibilityLabel = subscribed
+      .map { isSubscribed in
+        isSubscribed
+          ? localizedString(key: "Unsubscribe", defaultValue: "Unsubscribe")
+          : Strings.Subscribe()
+    }
+
+    configData
+      .takePairWhen(isSubscribedEvent.values())
+      .observeValues { configData, isSubscribed in
+        let (project, liveStream, _) = configData
+        AppEnvironment.current.koala.trackLiveStreamToggleSubscription(project: project,
+                                                                       liveStream: liveStream,
+                                                                       subscribed: isSubscribed)
+    }
+  }
+
+  private let configData = MutableProperty<(Project, Project.LiveStream, LiveStreamEvent?)?>(nil)
+  public func configureWith(project: Project, liveStream: Project.LiveStream, event: LiveStreamEvent?) {
```

Comment:
feel like this could be an enum such as `LiveStreamData` then you could use dot notation to access what you need.